builtin/providers/aws/data_source_aws_elastic_beanstalk_solution_stack.go

```diff
@@ -0,0 +1,109 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/service/elasticbeanstalk"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsElasticBeanstalkSolutionStack() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsElasticBeanstalkSolutionStackRead,
+
+		Schema: map[string]*schema.Schema{
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSolutionStackNameRegex,
+			},
+			"most_recent": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			// Computed values.
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// dataSourceAwsElasticBeanstalkSolutionStackRead performs the API lookup.
+func dataSourceAwsElasticBeanstalkSolutionStackRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticbeanstalkconn
+
+	nameRegex, nameRegexOk := d.GetOk("name_regex")
+
+	if !nameRegexOk {
+		return fmt.Errorf("name_regex must be assigned")
+	}
+
+	var params *elasticbeanstalk.ListAvailableSolutionStacksInput
+
+	resp, err := conn.ListAvailableSolutionStacks(params)
+	if err != nil {
+		return err
+	}
+
+	var filteredSolutionStacks []*string
+
+	r := regexp.MustCompile(nameRegex.(string))
+	for _, solutionStack := range resp.SolutionStacks {
+		if r.MatchString(*solutionStack) {
+			filteredSolutionStacks = append(filteredSolutionStacks, solutionStack)
+		}
+	}
+
+	var solutionStack *string
+	if len(filteredSolutionStacks) < 1 {
+		return fmt.Errorf("Your query returned no results. Please change your search criteria and try again.")
+	}
+
+	if len(filteredSolutionStacks) > 1 {
+		recent := d.Get("most_recent").(bool)
+		log.Printf("[DEBUG] aws_elastic_beanstalk_solution_stack - multiple results found and `most_recent` is set to: %t", recent)
+		if recent {
+			solutionStack = mostRecentSolutionStack(filteredSolutionStacks)
+		} else {
+			return fmt.Errorf("Your query returned more than one result. Please try a more " +
+				"specific search criteria, or set `most_recent` attribute to true.")
+		}
+	} else {
+		// Query returned single result.
+		solutionStack = filteredSolutionStacks[0]
+	}
+
+	log.Printf("[DEBUG] aws_elastic_beanstalk_solution_stack - Single solution stack found: %s", *solutionStack)
+	return solutionStackDescriptionAttributes(d, solutionStack)
+}
+
+// Returns the most recent solution stack out of a slice of stacks.
+func mostRecentSolutionStack(solutionStacks []*string) *string {
+	return solutionStacks[0]
```

Comment:
is this guaranteed to be the most recent? Don't use need to sort these using a date or something